### PR TITLE
ARTEMIS-468 Fix openwire redelivery related regressions under integration-tests

### PR DESCRIPTION
**Pull request by gaohoward**

### CODE REVIEW

**Comment by gaohoward** [comment]
look like Jenkins has a memory problem.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/ConsumerListener.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.core.server;
+
+/**
+ */
```
Comment (clebertsuconic) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/ConsumerListener.java]: I don't like this...

We have the SessionCallback to support protocol specific things.

the protocolData should be accessed by Protocol specific classes only only. We shouldn't be doing any sort of weird conversions here. 

I'm trying to keep the implementation clean and easy to understand.

Add a method on SessionCallback:

canceledDelivery(ServerConsumer, MessageReference, boolean failed);

And simply call:

callback.canceledDelivery(this, ref, failed) from ServerConsumer where you were calling ConsumerListener,


And remove this interface please.

File: artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/amq/AMQConsumer.java [hunk 2]
```diff
@@ -303,6 +307,22 @@ public void setPrefetchSize(int prefetchSize) {
...
 
+   @Override
+   public void updateForCanceledRef(MessageReference ref) {
+      long seqId = ref.getMessage().getMessageID();
+      long lastDelSeqId = info.getLastDeliveredSequenceId();
+      ServerMessage coreMessage = ref.getMessage();
+      int redeliveryCounter = coreMessage.getIntProperty(OpenWireMessageConverter.AMQ_MSG_REDELIVER_COUNTER);
+      if (openwireDestination.isTopic()) {
+         redeliveryCounter++;
+         coreMessage.putIntProperty(OpenWireMessageConverter.AMQ_MSG_REDELIVER_COUNTER, redeliveryCounter);
```
Comment (clebertsuconic) [inline artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/amq/AMQConsumer.java]: This is wrong anyways Howard, The same message could be delivered to another Queue (another topic for instance).

DeliveryCount is a property of the MessageReference, not message.

**Comment by clebertsuconic** [comment]
@gaohoward  I have merged with my commit ammending this. if you could take a look please?

if you need to amend the rule at least we have a single place to do that now.

**Pull request closed without merging**